### DISCUSSION TITLE
Help me understand relationships in budibase vs. classic relational model

### DISCUSSION THREAD

**Discussion by MrWombastic** [Budibase/budibase#8083] [Help] [answered]
Hi,

I am not sure if I understand anything correctly regarding relationships in budibase.
I am coming from MS SQL and Access world and I search for a tool to eliminate MS Access. So what I need is a well looking front end with integrations to MS SQL and the possibility to start Queries on the SQL Server or do complex calculations and automations directly in budibase.

For my PoC i started with a cloud instance and I am using the internal db. But I am getting mad about relationsships.

For example:
Table Transactiondetails has several columns which are relationships (AssetID -> realted on an AssetTable, TransactionID -> realted on a header table with header information about the transaction,..).

Now if I want to get a sum of all of my transactions grouped by the asset, it's simply NOT possible.

In the view of the transactiondetails table , I am not able to group by relationship fields. It only offers Numbers, Dates and plain Text.
![image](https://user-images.githubusercontent.com/34077172/193409543-5b223659-6a9f-4323-b951-4b166727b9f4.png)

If i try it by the assets table which is connected to every single row of the transactiondetails there seems to be no way to sum up the value of all referenced transactions, because a formula column can only be done per row.

So I really need to ask mysef if I misunderstand relationships completly or if it's simply completly different from MS SQL / Access.

**Comment by andz-bb** [comment, marked as answer]
hey @MrWombastic 

the internal Budibase DB uses CouchDB which handles querying and relationships very differently to how MS SQL would. It's a document-based non-relational database. This means that relationships look a bit different to what you're used to on SQL Server and there are limitations on what you can do vs. a more traditional relational db.

It's worth having a look at [our documentation on Budibase DB Relationships.](https://docs.budibase.com/docs/relationships) There's a good explanation of the differences between the available db options and a demo of using Postgres for relationships [in this video here.](https://vimeo.com/719112585)

It could be easier for you to use MS SQL with Budibase if that's what you're used to and we have full support for it in the Budibase builder. If you have any specific questions, let me know and I'll be happy to help. 👍
